jiant/tasks/lm.py

```diff
@@ -246,16 +246,25 @@ def update_metrics(self, out, batch=None):
         return
 
     def get_data_iter(self, path):
-        """Loading data file and tokenizing the text
+        """
+        Loading data file and tokenizing the text. We treat the Wikipedia corpus as a
+        long sequence, and we take each slice of 510 tokens as an example. The dataset
+        consists of a sentence per row in the file. This function concatenates all of the sentences,
+        before going through the sequence and yielding each chunk of 510 tokens.
         Args:
             path: (str) data file path
         """
+        seq_len = self.max_seq_len - 2
+        total_tokens = []
         with open(path, "r", encoding="utf-8") as txt_fh:
             for row in txt_fh:
                 toks = row.strip()
                 if not toks:
                     continue
-                yield tokenize_and_truncate(self._tokenizer_name, toks, self.max_seq_len)
+                total_tokens += toks
```

Comment:
This change has me confused* (so I kicked off runs to compare the behavior of this code pre/post change):
* pre-change `tokenize_and_truncate()` was called and `sent_` (at L270) was a list of word-level tokens.
* post-change `tokenize_and_truncate()` is not called, and instead `total_tokens += toks` is performed, and this results in a list of characters (and later on `sent_` (at L279) is also list of characters).

Is the move from word- to character-level tokens intentional?

\* please let me know if there's other context I'm overlooking that would help explain this change.

Reply:
You are completely correct. I caught this before we ran the full set of K sweeps yesterday, but we were head over heels with other things so didn't update this PR yesterday.

Reply:
Let me push now what we're currently using.